### PR TITLE
Port ocp4 to Python

### PR DESCRIPTION
**Pull request by locriandev**
Ref. https://issues.redhat.com/browse/ART-3672

Test runs:
- [4.12, no source changes](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/dpaolell/job/ocp4/141/console)
- [4.14, forced builds, 1 RPM](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/dpaolell/job/ocp4/145/console)
- [4.12 forced mass rebuild](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/dpaolell/job/ocp4/142/console)
- [4.14, 7 images, no RPMs](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/dpaolell/job/ocp4/152/console). Triggered [build-sync](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/aos-cd-builds/job/build%252Fbuild-sync/33741/) and [olm-bundle](https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/aos-cd-builds/job/build%252Folm_bundle/9518/)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,66 +0,0 @@
-properties(
```
Comment (joepvd) [inline Jenkinsfile:1]: This file is important.

Comment (locriandev) [reply Jenkinsfile]: Thanks for noticing, it was a hack space workaround that was meant to be temporary. Reverted

File: pyartcd/pyartcd/exectools.py [hunk 2]
```diff
@@ -106,3 +106,15 @@ async def cmd_assert_async(cmd: Union[List[str], str], check: bool = True, **kwa
         else:
             logger.warning(msg)
     return proc.returncode
+
+
+async def retry_async(cmd: Union[List[str], str], max_retries: int, **kwargs):
```
[the remark this answers is not in the document]
Comment (locriandev) [reply pyartcd/pyartcd/exectools.py:111]: Cool, I wasn't aware of that. I'd like to use it, but it seems we can only apply that decorator to a whole function? Here I am running two distinct commands within the same function, and I wouldn't want to re-execute the first one if the second one fails. Is there a more scoped way of using this library? Like a context manager...

[the remark this answers is not in the document]
Comment (locriandev) [reply pyartcd/pyartcd/exectools.py]: I proposed to use 2 small embedded functions: f7666fc4a2ad18fcadb29a868dcb4d33f4d8ac03

[the remark this answers is not in the document]
Comment (locriandev) [reply pyartcd/pyartcd/exectools.py]: I proposed to use 2 small embedded functions: https://github.com/openshift-eng/aos-cd-jobs/commit/f7666fc4a2ad18fcadb29a868dcb4d33f4d8ac03

File: pyartcd/pyartcd/util.py [hunk 3]
```diff
@@ -310,3 +310,232 @@ async def sync_images(version: str, assembly: str, operator_nvrs: list,
...
         )
+
+
+def default_release_suffix():
+    """
+    Returns a release suffix based on current timestamp
+    E.g. "201901011200.?
+    """
+
+    return f'{datetime.strftime(datetime.now(), "%Y%m%d%H")}.p?'
```
Comment (vfreex) [inline pyartcd/pyartcd/util.py:321]: The timestamp should include minute:

```suggestion
    return f'{datetime.strftime(datetime.now(), "%Y%m%d%H%M")}.p?'
```

File: pyartcd/pyartcd/util.py [hunk 4]
```diff
@@ -310,3 +310,232 @@ async def sync_images(version: str, assembly: str, operator_nvrs: list,
             doozer_data_path=doozer_data_path,
             doozer_data_gitref=doozer_data_gitref
         )
+
+
+def default_release_suffix():
+    """
+    Returns a release suffix based on current timestamp
+    E.g. "201901011200.?
```
Comment (vfreex) [inline pyartcd/pyartcd/util.py:318]: 201901011200.? -> 201901011200.p?

File: pyartcd/pyartcd/exectools.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (locriandev) [reply pyartcd/pyartcd/exectools.py]: thanks, I've applied this pattern in 1307c0843e42c41090e775764b3b6b2445ba0a5c

File: pyartcd/pyartcd/util.py (hunk 4 above)
Comment (locriandev) [reply pyartcd/pyartcd/util.py:318]: Fixed in 047f1e932b5cb53b464f24227ee324c30a9caec0

File: pyartcd/pyartcd/util.py (hunk 3 above)
Comment (locriandev) [reply pyartcd/pyartcd/util.py]: Fixed in a82d6f286dc689606ee41ebb42e9d87fc666a59c

**Review by vfreex — APPROVED** [review]
LGTM

**Review by joepvd — APPROVED** [review]
/lgtm

**Pull request merged**